helm/openwhisk/templates/invoker-agent-pod.yaml

```diff
@@ -1,7 +1,7 @@
 # Licensed to the Apache Software Foundation (ASF) under one or more contributor
 # license agreements; and to You under the Apache License, Version 2.0.
 
-{{ if .Values.invoker.containerFactory.kubernetes.agent.enabled }}
+{{ if (.Values.invoker.containerFactory.kubernetes.agent.enabled) and not .Values.controller.lean }}
```

Comment:
I removed it as it's not mandatory, but in case user specifies in lean mode:
```invoker:
  containerFactory:
    kubernetes:
      agent:
        enabled: true
```
it will spawn unnecessary invoker-agent pods